int16 ID
string action
int32 duration
